lib_eio/pool.mli

```diff
@@ -0,0 +1,24 @@
+type 'a t
+
+type 'a handlers = {
+  check : 'a -> bool;
+  dispose : 'a -> unit;
+}
+
+val noop_handlers : 'a handlers
+
+val create :
+  sw:Switch.t ->
+  alloc:(unit -> 'a * 'a handlers) ->
+  int ->
+  'a t
+
+val use : 'a t -> ('a -> 'b) -> 'b
+
+val async : sw:Switch.t -> 'a t -> ('a -> unit) -> unit
```

Comment:
Do you think these are necessary (i.e. do they need to be exposed to the user, I realise they're used internally) ? In my mind, the `use` function is the main one and I can decide to fork the fiber that calls it should I need to. Otherwise we've kinda added `fork` and `fork_promise` with two slightly different names (using `async`) which feels a little cluttered ?

Reply:
Ah good point. Leaving the forking responsibilities to users gives a cleaner picture yes. They did feel a bit sloppy and redundant when I was writing it.